velox/functions/sparksql/specialforms/FromJson.cpp

```diff
@@ -99,7 +101,63 @@ struct ExtractJsonTypeImpl {
   struct KindDispatcher<TypeKind::INTEGER, Dummy> {
     static simdjson::error_code
     apply(Input value, exec::GenericWriter& writer, bool /*isRoot*/) {
-      return castJsonToInt<int32_t>(value, writer);
+      if (writer.type() == DATE()) {
+        SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
+        std::string_view s;
+        switch (type) {
+          case simdjson::ondemand::json_type::string: {
+            SIMDJSON_ASSIGN_OR_RAISE(s, value.get_string());
+            break;
+          }
+          default:
+            return simdjson::INCORRECT_TYPE;
+        }
+        int32_t day = 0;
```

Comment:
Can we extract the convertStringToDate logic to a new function, looks like we may
 support other conversions in the future, here will be much logic

Reply:
Updated, thanks.